src/core/include/openvino/core/type.hpp

```diff
@@ -81,20 +81,28 @@ namespace frontend {
 class ConversionExtensionBase;
 }  // namespace frontend
 
+namespace compile {
 template <typename T>
 constexpr bool use_ov_dynamic_cast() {
     return std::is_base_of_v<ov::frontend::ConversionExtensionBase, T>;
```

Comment:
right now use_ov_dynamic_cast depends only on ConversionExtensionBase which is specific to frontends. Can this check be moved to frontends common part?

Reply:
That'd imply dependency of almost all the objects to frontend/common, mostly not needed but affecting build time.